components/client/src/main/java/com/hotels/styx/client/netty/connectionpool/NettyToStyxResponsePropagator.java

```diff
@@ -166,17 +155,8 @@ protected void channelRead0(ChannelHandlerContext ctx, Object msg) throws Except
 
     @Override
     public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
-        FlowControllingHttpContentProducer producer = getContentProducer(ctx);
-
         if (evt instanceof IdleStateEvent) {
-            producer.channelInactive(
-                    new ResponseTimeoutException(
-                            origin,
-                            "idleStateEvent",
-                            producer.receivedBytes(),
-                            producer.receivedChunks(),
-                            producer.emittedBytes(),
-                            producer.emittedChunks()));
+            getContentProducer(ctx).tearDownResources("idle timeout");
```

Comment:
It would better to just pass a cause to `tearDownResources`. Then it would be possible to pass in a `RequestTimeoutException` or `ContentTimeoutException` where appropriate.